### ISSUE TITLE
Control TFTS/TAAFT truncation by cutoff frequency/period instead of frequency spectrum ratio

### ISSUE THREAD

**Issue by kahaaga** [JuliaDynamics/TimeseriesSurrogates.jl#54]
Dropping an idea, so that I don't forget it:

Currently, the cutoff is defined in terms of the ratio of the low-/high-frequency domain to the whole frequency domain (which is what they do in the Nakamura et al. paper). It would be user-friendly if the frequency threshold for the truncated Fourier surrogates family could be specified using an actual frequency or period. 

Maybe define a simple 

```julia
struct FrequencyCutoff
    fϵ::Union{Real, Tuple{Int, Int}}
end

struct PeriodCutoff
    fϵ::Union{Real, Tuple{Int, Int}}
end
```

Depending on the context, the user can then decide to filter frequencies above/below some `FrequencyCutoff`, or filter oscillations that are slower/faster than `PeriodCutoff`. If the cutoffs are a `(low, high)` tuple, then frequencies/periods within some range are kept, while the rest are randomized.

These types could be applied to any of the truncated Fourier transform-based surrogate methods (until now, truncated version of `RandomFourier` and `AAFT` are implemented in #47, but this will be extended to IAAFT and possibly other variants of the AAFT algorithm such as CAAFT). 

Any thoughts?

**Comment by Datseris** [comment]
I think this adds a lot of complication without gain. There is no reason to support both period and frequency....

I am also not sure whether this is worth creating structs. What is the gain from having simple arguments?

**Comment by kahaaga** [comment]
The gain is having a conceptually meaningful argument to `TFTS`. "Fraction of frequency spectrum", which Nakamura et al. use in their paper, is difficult to think of intuitively. "Frequencies below/above 1/100" are more meaningful. 

We don't need to use the exact approach in the original paper, so I could just rephrase the argument to `TFTS` be a cutoff frequency instead of indicating a portion of the spectrum. That achieves the same as the structs would.

**Comment by Datseris** [comment]
It is possible to allow two arguments, only one of which must be specified
```julia
f(a1=nothing,a2=nothing)
a1==a2==nothing && error("provide at least 1 arg")
a!=nothing && a2!=nothing && error("must provide only 1 arg")
```
this way you can keep the original ratio-based definition and also add the more sensible frequency-specified argument.

**Comment by kahaaga** [comment]
I like that. Then

- `TFTS(fϵ)` with `fϵ ∈ [-1, 0) ∪ (0, 1]` uses the ratio based definition (randomizing low frequencies if `fϵ < 0` and randomizing high frequencies if `fϵ > 0`.
- `TFTS(fϵ_lo, fϵ_hi)` with `fϵ_lo, fϵ_hi` indicates that a range of frequencies should be preserved. We can then use the convention: `(fϵ_lo, fϵ_hi) ∈  [-1, 0)` randomizes frequencies *within* the specified frequency range, while  `(fϵ_lo, fϵ_hi) ∈  (0, 1]` randomizes frequences *outside* the specified frequency range. 

If you're ok with that, I'll make a PR adding the two-argument option.

**Comment by Datseris** [comment]
I don't like the usage of plus/minus. It is not only unintuitive for me, but extremely difficult to memorize.

I think something like

```
TFTS(fε::Real, low::Bool)
```
is more suitable for the first version, where the `low` denotes the cutoff.

Similarly for second case:
```
TFTS(fε::NTuple{2, Real}, within::Bool)
```
where the `within` denotes the inside/outside. Mentioning that `Inf` can be part of the tuple is also useful, which in this way defines the cutoff point as a single frequency in some sense. (i.e. `(0.5, Inf)` means all frequencies above 0.5.

Important to clarify withether the elements of the tuple are included in the range or not: does `(0.5, Inf)` mean frequencies in $[0.5, \infty)$  or in $(0.5, \infty)$ ?
